scipy/optimize/optimize.py

```diff
@@ -425,8 +424,26 @@ def _minimize_neldermead(func, x0, args=(), callback=None,
         ``initial_simplex[j,:]`` should contain the coordinates of
         the j-th vertex of the ``N+1`` vertices in the simplex, where
         ``N`` is the dimension.
-
+    xatol : float, optional
+        Absolute error in xopt between iterations that is acceptable for
+        convergence.
+    fatol : number, optional
+        Absolute error in func(xopt) between iterations that is acceptable for
+        convergence.
     """
+    if 'xtol' in unknown_options:
+        xatol = unknown_options['xtol']
+        warnings.warn("xtol is deprecated for _minimize_neldermead,"
+                      " use xatol instead.",
+                      DeprecationWarning)
+        unknown_options.pop('xtol')
+    if 'ftol' in unknown_options:
+        fatol = unknown_options['ftol']
+        unknown_options.pop('ftol')
+        warnings.warn("ftol is deprecated for _minimize_neldermead,"
+                      " use fatol instead.",
+                      DeprecationWarning)
+
```

Comment:
Perhaps I can add a `deprecated_kwds` decorator to `_minimize_neldermead`, something along the lines of:

    def deprecated_kwds(f, kwds_to_deprecate, deprecation_message):
        def deprecater_func(*args, **kwds):
            # see if any of kwds_to_deprecate are in kwds, if so filter/raise warning
            # the decorator will also know if both fatol and ftol have been supplied.
            # so I can make a choice on which one to use.
            return f(*args, **kwds)
        return deprecater_func
       
This could test if ftol and fatol are being supplied. If they are, then I can raise a warning. Alternatively I could simply test if the values for ftol and fatol are the default value. If not, then I select the one that's been changed.
I think the decorator approach is probably a little more robust. I can see that it would be useful elsewhere as well.